src/content/blog/astro-460.mdx

```diff
@@ -0,0 +1,116 @@
+- [**Ability to move the dev toolbar**](#ability-to-move-the-dev-toolbar)
+- [**Experimental: Support for CSRF protection**](#experimental-support-for-csrf-protection)
+- [**Cookies improvements**](#cookies-improvements)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Deprecated support for older versions of Node
+
+In accordance with [our Node support and upgrade policy](https://docs.astro.build/en/upgrade-astro/#nodejs-support-and-upgrade-policies), this version of Astro deprecate support for versions of Node lower than 18.17.1, Node 19 and versions of Node lower than 20.3.0. As always, this decision wasn't reached lightly, and we believe this change will result in a more stable and secure Astro experience for all users.
+
+Astro will still work on these older versions of Node until Astro 5, however you may see a warning when installing Astro and certain features may not work as expected.
+
+## Manual routing strategy for internationalization
+
+Astro 4.6 introduces a new manual routing strategy for internationalization. This new strategy allows you to take total control over the routing of your internationalized Astro website, for cases where the default routing strategy doesn't quite meet your needs.
+
+To enable manual routing, set the `i18n.strategy` option in your `astro.config.mjs` to `manual`:
+
+```js
+// astro.config.mjs
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  i18n: {
+    locales: ["en", "fr"],
+    defaultLocale: "fr",
+    routing: "manual",
+  },
+});
+```
+
+and then [add a middleware](https://docs.astro.build/en/guides/middleware/) to your project to handle the routing:
+
+```ts
+// src/middleware.ts
+import { redirectToDefaultLocale } from "astro:i18n";
+
+// Example middleware that redirects all requests to the default locale apart for the /about page
+export const onRequest = defineMiddleware(async (context, next) => {
+  if (context.url.startsWith("/about")) {
+    return next();
+  } else {
+    return redirectToDefaultLocale(context, 302);
+  }
+});
+```
+
+Check out the [internationalization documentation](https://docs.astro.build/en/guides/internationalization/) for more information on how to use manual routing with internationalization.
+
+## Ability to move the dev toolbar
+
+Astro 4.6 introduces the ability to move the dev toolbar to different positions around the screen. This can be useful if you have a sticky header that covers the bottom of the screen, or if you just prefer the toolbar to be somewhere else. Thanks to [Ming-jun Lu](https://github.com/mingjunlu) for contributing this feature!
+
+## Experimental: Support for CSRF protection
```

Comment:
This has also changed, and the changeset has the up to date description and example! Also happy for Erika to revise based on this, and will look again afterwards!